src/lib/retry-logic.ts

```diff
@@ -0,0 +1,71 @@
+import { AxiosError, AxiosRequestConfig } from 'axios';
+import isRetryAllowed from 'is-retry-allowed';
+import { UserPayload } from '../services/user-payload';
+import { ChtApi } from './cht-api';
+
+const RETRY_COUNT = 4;
+
+export const axiosRetryConfig = {
+  retries: RETRY_COUNT,
+  retryDelay: () => 1000,
+  retryCondition: (err: AxiosError) => {
+    const status = err.response?.status;
+    return (!status || status >= 500) && isRetryAllowed(err);
```

Comment:
Maybe we should directly target the status codes we know we can potentially recover from? Like the list in https://github.com/medic/cht-user-management/issues/17 503s, 409s

Reply:
Sure. 

`!status && isRetryAllowed(err)` I beleive is going to be mostly timeouts
`>= 500` can recover from 500, 502, 503, 504, 511.  As mentioned above, we can't handle 409s generally here.